db/migrations/versions/~2024_10_17_0818-de8d7954ffda_add_funding_type_to_the_fund_table.py

```diff
@@ -0,0 +1,47 @@
+"""Add funding type to the fund table
+
+Revision ID: de8d7954ffda
+Revises: f5d2a21f372e
+Create Date: 2024-10-17 08:18:08.965508
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "de8d7954ffda"
+down_revision = "f5d2a21f372e"
+branch_labels = None
+depends_on = None
+
+# it looks like alembic won't autogenerate the enum type with postgres, we can explicitly prompt it
+# to create and drop during upgrade and downgrade
+# https://github.com/sqlalchemy/alembic/issues/278
+funding_type_enum = sa.Enum("COMPETITIVE", "UNCOMPETED", "EOI", name="fundingtype")
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+
+    # make sure we've got the type before using it in the new column
+    funding_type_enum.create(op.get_bind(), checkfirst=True)
+
+    with op.batch_alter_table("fund", schema=None) as batch_op:
+        batch_op.add_column(
+            sa.Column(
+                "funding_type",
+                funding_type_enum,
+                nullable=True,
```

Comment:
As per my reply to Nuwan, can we create a nullable, then backfill, then make not null all in one go

Reply:
https://github.com/communitiesuk/funding-service-design-fund-store/pull/342/commits/edb2ae84e3d1d62c57b85b67e28c0f394f52d981